app/controllers/api/physical_storages_controller.rb

```diff
@@ -20,6 +20,21 @@ def create_resource(_type, _id = nil, data = {})
       action_result(false, err.to_s)
     end
 
+    def delete_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" if id.blank?
+
+      api_action(type, id) do |klass|
+        physical_storage = resource_search(id, type, klass)
+        raise physical_storage.unsupported_reason(:delete) unless physical_storage.supports?(:delete)
+        msg = "Detaching #{physical_storage_ident(physical_storage)}"
+        api_log_info(msg)
+        task_id = physical_storage.delete_physical_storage_queue(User.current_user)
+        action_result(true, msg, :task_id => task_id)
+      rescue => err
```

Comment:
@llivne I believe this is causing your codeclimate issues, try adding a `begin` around this like
```
api_action(type, id) do |klass|
  begin
    physical_storage = resource_search(id, type, klass)
    etc...
  rescue => err
    action_result(false, err.to_s)
  end
end
```

Reply:
You were right. fixed

Reply:
That's weird... do/rescue was introduced in Ruby 2.5, so it _should_ work

Reply:
Yeah codeclimate seems to still be checking ruby 2.4 syntax even though rubocop isn't.